`lotus-miner proving compute window-post` cmd panics with `index out of range`

### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus fvm/fevm - Lotus FVM and FEVM interactions
- [ ] lotus miner/worker - sealing
- [X] lotus miner - proving(WindowPoSt/WinningPoSt)
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
lotus-miner version
Daemon:  1.21.0-rc1+mainnet+git.3f8a41cfe+api1.5.0
Local: lotus-miner version 1.21.0-rc1+mainnet+git.3f8a41cfe
```


### Repro Steps

One day windowpost started failing on a partition.
`lotus-miner proving check --slow` does not reveal any bad sectors.
But the actual window post fails with panic.

### Describe the Bug

```
lotus-miner proving check --slow --only-bad 12
deadline  partition  sector  status
```
Nothing found.

Simulated windowpost on the partition but it failed with a panic.

```
lotus-miner proving compute window-post 12
Took 6m24.691367409s
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.glob..func56(0xc0000eba40)
	/home/downloads/lotus/cmd/lotus-miner/proving.go:662 +0x8ff
github.com/urfave/cli/v2.(*Command).Run(0x793b160, 0xc0000eb980)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:173 +0x6ca
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc0001a2e00, 0xc0000eb840)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:531 +0xd85
github.com/urfave/cli/v2.(*Command).startApp(0x793b280, 0xc0000eb840)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:291 +0x79b
github.com/urfave/cli/v2.(*Command).Run(0xc0000eb840?, 0xc000267040?)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:95 +0xba
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc0001a2c40, 0xc0000eb5c0)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:531 +0xd85
github.com/urfave/cli/v2.(*Command).startApp(0x793b4c0, 0xc0000eb5c0)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:291 +0x79b
github.com/urfave/cli/v2.(*Command).Run(0x442ff40?, 0xc000055980?)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:95 +0xba
github.com/urfave/cli/v2.(*App).RunContext(0xc0004e8c40, {0x594a5b0?, 0xc000056d60}, {0xc0000520a0, 0x5, 0x5})
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:383 +0xfde
github.com/urfave/cli/v2.(*App).Run(...)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:252
github.com/filecoin-project/lotus/cli.RunApp(0x4564060?)
	/home/downloads/lotus/cli/helper.go:48 +0x107
main.main()
	/home/downloads/lotus/cmd/lotus-miner/main.go:168 +0x115a

```

Checked lotus-miner log, a sector was being skipped:

`2023-03-26T19:29:35.374+0900	INFO	wdpost	wdpost/wdpost_run.go:397	running window post	{"cycle": "2023-03-26T19:26:15.045+0900", "chain-random": "5PO3wNmwroeDFZwJeREQtYothJluaqiEduleI714ch4=", "deadline": {"CurrentEpoch":2717332,"PeriodStart":2715091,"Index":12,"Open":2717311,"Close":2717371,"Challenge":2715791,"FaultCutoff":2717241,"WPoStPeriodDeadlines":48,"WPoStProvingPeriod":2880,"WPoStChallengeWindow":60,"WPoStChallengeLookback":20,"FaultDeclarationCutoff":70}, "height": "2717332", "skipped": 1}`


Removed and terminated the skipped sectors, and retried lotus-miner proving compute windowpost.

This time nothing is being skipped:
`2023-03-26T19:44:24.253+0900	INFO	wdpost	wdpost/wdpost_run.go:412	computing window post	{"cycle": "2023-03-26T19:38:27.605+0900", "batch": 0, "elapsed": 356.518975334, "skip": 0, "err": null}`

But window post still fails:

```
lotus-miner proving compute window-post 12
Took 5m56.707843203s
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.glob..func56(0xc000394b40)
	/home/downloads/lotus/cmd/lotus-miner/proving.go:662 +0x8ff
github.com/urfave/cli/v2.(*Command).Run(0x793b160, 0xc000394a40)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:173 +0x6ca
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000150c40, 0xc000394740)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:531 +0xd85
github.com/urfave/cli/v2.(*Command).startApp(0x793b280, 0xc000394740)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:291 +0x79b
github.com/urfave/cli/v2.(*Command).Run(0xc000394740?, 0xc00041ee10?)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:95 +0xba
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000150a80, 0xc0003941c0)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:531 +0xd85
github.com/urfave/cli/v2.(*Command).startApp(0x793b4c0, 0xc0003941c0)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:291 +0x79b
github.com/urfave/cli/v2.(*Command).Run(0x442ff40?, 0xc000054980?)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:95 +0xba
github.com/urfave/cli/v2.(*App).RunContext(0xc000286e00, {0x594a5b0?, 0xc000138c70}, {0xc00012c000, 0x5, 0x5})
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:383 +0xfde
github.com/urfave/cli/v2.(*App).Run(...)
	/root/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:252
github.com/filecoin-project/lotus/cli.RunApp(0x4564060?)
	/home/downloads/lotus/cli/helper.go:48 +0x107
main.main()
	/home/downloads/lotus/cmd/lotus-miner/main.go:168 +0x115a

```







### Logging Information

```text
-
```

--- a/cmd/lotus-miner/proving.go
+++ b/cmd/lotus-miner/proving.go
@@ -658,6 +658,10 @@ It will not send any messages to the chain.`,
 		for _, i := range res {
 			var postParam SubmitWindowedPoStParams
 			postParam.Deadline = i.Deadline
+
+			// Initialize the postParam.Partitions slice with the same length as i.Partitions
+			postParam.Partitions = make([]PoStPartition, len(i.Partitions))
+
 			for id, part := range i.Partitions {
 				postParam.Partitions[id].Index = part.Index
 				count, err := part.Skipped.Count()

